Automattic/jetpack
# Site Settings API: Fix saving subscription options

Fixes https://github.com/Automattic/wp-calypso/issues/74932

## Proposed changes:

* The `subscription_options` site option has a value of an associative array type. Currently, we only allow overriding the entire stored associative array if the API is provided with exactly two key-value pairs: invitation and comment_follow. If there is an additional key-value pair or if one of the listed ones is missing, the update request for the site option will be ignored. This PR introduces the ability to partially update the site option while retaining the old values for any keys that the site already stores under the subscription_options site option but has not included in the update request.

### Other information:


## Jetpack product discussion


p2EDhh-1UB-p2

## Does this pull request change what data or activity we track or use?



## Testing instructions:






### Simple Site Testing instructions
Prerequisits:
- a test, simple site where `subscription_options` site option contains at least one other key-value pair, than the standard: `invitation` and `comment_follow` options. An example of a common non-standard `subscription_options` key-value pair: `"loggedoutfollow" => "on"`

* Apply the changes to your sandbox
* Sandbox `public-api.wordpress.com`
* Go to https://wordpress.com/settings/reading of your test site
* Make changes to any of the **Confirmation email message** or the **Comment follow email message**, and press **Save settings**
* Reload the page, and ensure that the values in the **Confirmation email message** and the **Comment follow email message** text boxes still display the updated text
